central/virtualmachine/datastore/datastore_impl.go

```diff
@@ -7,13 +7,15 @@ import (
 	"github.com/pkg/errors"
 	"github.com/stackrox/rox/central/metrics"
 	virtualMachineStore "github.com/stackrox/rox/central/virtualmachine/datastore/internal/store"
+	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/protocompat"
 	"github.com/stackrox/rox/pkg/search"
 )
 
 const (
-	defaultResultSize = 1000
+	defaultPageSize = 100
+	preAllocateCap  = 65535
```

Comment:
nit:  Use this constant
https://github.com/stackrox/stackrox/blob/1f3073dcb87a415a0ff57dc8bd3752510ecbfbd0/pkg/search/query_builder.go#L44 OR at least `math.MaxUint16`.  Though I do wonder if we ever expect that many so perhaps the pre allocation cap should be much lower.